### DISCUSSION TITLE
navigateTo with route alias keeps creating an infinite loop

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#25531] [Questions]
I have a problem where I have a file structure that looks like this
```
/pages
  /_viewport
    /desktop
      /products
        /[handle].vue
    /mobile
      /products
        /[handle].vue
```

And what I want to do is if a user clicks on a link `/viewport/mobile/products/t-shirt` that there will be an alias of `/products/t-shirt`.

I have a middleware function setup but I don't know how to get this to work, I keep running into the same problem no matter what I do.
```js
export default defineNuxtRouteMiddleware(async (to, from) => {
  if (to.path === '/viewport/mobile/products/t-shirt') {
    return navigateTo({
      path: `/viewport/mobile/products/${to.params.product}`,
      alias: `/products/${to.params.product}`
    })
  }

})
```

If a user clicks `/viewport/mobile/products/t-shirt`, how can I match that path, and have the url shown as `/products/t-shirt`?

**Comment by undermoonn** [comment]
`alias` is a property of route, it should be defined in route component :)

```vue
// page/_viewport/mobile/products/[handle].vue
<script lang="ts" setup>
definePageMeta({
  alias: '/products/:handle',
})
</script>
```

```ts
export default defineNuxtRouteMiddleware((to, from) => {
  if (to.path === '/_viewport/mobile/products/t-shirt') {
    return navigateTo({
      path: `/products/${to.params.handle}`,
    })
  }
})
```

**Reply by undermoonn** [reply]
But you can probably not use same `alias` in `page/_viewport/mobile/products/[handle].vue` and `page/_viewport/desktop/products/[handle].vue`.

If you wan to, you can choose `redirect` instead of `alias`. And then move both two page component codes to `page/products/[handle].vue` and remove middleware.
